File: securescope-backend/securescope-backend/src/main/java/com/securescope/scanner/dto/ScanResult.java
package com.securescope.scanner.dto;

import com.securescope.common.enums.RiskLevel;
import com.securescope.common.enums.ScanStatus;
import java.util.List;

public record ScanResult(
	String scanId,
	ScanStatus status,
	int securityScore,
	RiskLevel riskLevel,
	int totalFindings,
	List<FindingResult> findings
) {
}

File: securescope-frontend/src/app/core/models/scan-result.ts
import { FindingResult } from './finding-result';

export interface ScanResult {
  scanId: string;
  status: 'PENDING' | 'RUNNING' | 'COMPLETED' | 'FAILED' | 'CANCELLED';
  securityScore: number;
  riskLevel: 'SAFE' | 'LOW' | 'MODERATE' | 'HIGH' | 'CRITICAL';
  totalFindings: number;
  findings: FindingResult[];
}
